### PR TITLE
Refactor/507

### PR DESCRIPTION
**Pull request by Haytham1337**
Refactored and improved user authentication with refresh token method.
image
local run tests suits
image

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: utils/rules.js [hunk 1]
```diff
@@ -11,9 +11,9 @@ const {
   STATUS_CODES: { FORBIDDEN, UNAUTHORIZED },
 } = require('../consts/status-codes');
 
-const isAuthorized = rule()((parent, args, context, info) =>
-  context.user ? true : new RuleError(USER_NOT_AUTHORIZED, UNAUTHORIZED)
-);
+const isAuthorized = rule()((parent, args, context, info) => {
+  return context.user ? true : new RuleError(USER_NOT_AUTHORIZED, UNAUTHORIZED);
+});
```
Comment (AndriyLymych) [inline utils/rules.js:16]: please, do inline func

File: utils/create-tokens.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken');
+const { REFRESH_TOKEN_EXPIRES_IN, SECRET } = require('../dotenvValidator');
+
+const generateTokens = (userId, params, withRefresh = false) => {
+  const options = {
+    expiresIn: params.expiresIn,
+  };
+
+  if (withRefresh) {
+    const accesToken = jwt.sign({ userId }, params.secret, options);
```
Comment (AndriyLymych) [inline utils/create-tokens.js:10]: put options into func, without creation const options

File: utils/rules.js (hunk 1 above)
Comment (Haytham1337) [reply utils/rules.js]: fixed

File: utils/create-tokens.js (hunk 2 above)
Comment (Haytham1337) [reply utils/create-tokens.js]: fixed

**Review by AndriyLymych — APPROVED** [review]

**Review by Kiyooooko — APPROVED** [review]

**Review by RevylaV — COMMENTED** [review]
Good job

**Pull request merged**